### ISSUE TITLE
[Example] support fuse batch normalization in JavaScript

### ISSUE THREAD

**Issue by huningxin** [intel/webml-polyfill#240]
Support fuse batch normalization in JavaScript when loading models.

**Comment by huningxin** [comment]
Found this one: https://tkv.io/posts/fusing-batchnorm-and-conv/

**Comment by pinzhenx** [comment]
Now I supported three fusing patterns in a regular convolution.
```
regular conv -> bn 
regular conv -> relu
regular conv -> bn -> relu
```

However, things can get complicated when a conv node could also be a group conv or a depthwise conv (special case of group conv):
- For group conv, we do not currently support this op.
- For depthwise conv, we cannot fold the following batch norm into it. Moreover, some batch norm nodes are not preceded by a conv node (e.g. resnet152v2).
  Thus, I added a separate case to handle batch norm using identity conv. Meanwhile, it will also look ahead to fuse the relu node. E.g.,
   - `bn` in resnet152v2 will be a separate node
   - `bn -> relu` in resnet152v2 will be a separate node
   - `d-conv -> bn -> relu` in mobilenetv2 will be converted to two nodes `d-conv -> (bn -> relu)`

**Comment by huningxin** [comment]
Your plan sounds good. Thanks!

**Comment by pinzhenx** [comment]
I managed to fuse batch norm into depthwise conv. So all batch norm preceded by a (regular/depthwise) conv now can be fused.

BTW, I found that the pytorch implementation of fusing batch norm given in the link is buggy.

```python
def fuse_conv_and_bn(conv, bn):
    ...
    w_conv = conv.weight.clone().view(conv.out_channels, -1)
    w_bn = torch.diag(bn.weight.div(torch.sqrt(bn.eps+bn.running_var)))
    fusedconv.weight.copy_( torch.mm(w_bn, w_conv).view(fusedconv.weight.size()) )

    if conv.bias is not None:
        b_conv = conv.bias
    else:
        b_conv = torch.zeros( conv.weight.size(0) )
    b_bn = bn.bias - bn.weight.mul(bn.running_mean).div(torch.sqrt(bn.running_var + bn.eps))
    fusedconv.bias.copy_( b_conv + b_bn )   // wrong!
    fusedconv.bias.copy_( bn.weight.div(torch.sqrt(bn.eps+bn.running_var).mul(b_conv) + b_bn ) // correct

    return fusedconv
```
